lib/appium.js

```diff
@@ -176,15 +178,26 @@ class AppiumDriver extends BaseDriver {
       }
     }
 
-    let curSessions;
+    let runningDriversData;
     try {
-      curSessions = this.curSessionDataForDriver(InnerDriver);
+      runningDriversData = this.curSessionDataForDriver(InnerDriver);
     } catch (e) {
       throw new errors.SessionNotCreatedError(e.message);
     }
 
+    let innerSessionId, dCaps;
     let d = new InnerDriver(this.args);
-    let [innerSessionId, dCaps] = await d.createSession(caps, reqCaps, curSessions);
+    try {
+      if (_.isArray(this.pendingDrivers[InnerDriver.name])) {
+        this.pendingDrivers[InnerDriver.name].push(d);
+      } else {
+        this.pendingDrivers[InnerDriver.name] = [d];
+      }
+      const otherPendingDriversData = this.pendingDrivers[InnerDriver.name].filter((drv) => d !== drv).map((drv) => drv.driverData);
+      [innerSessionId, dCaps] = await d.createSession(caps, reqCaps, [...runningDriversData, ...otherPendingDriversData]);
```

Comment:
For now we only return an empty object as session data and what we really care about is count of items in this array.
Nevertheless, if we really need to know more details for each driver type then we could simply include this information into _driverData_ structure itself